packages/upset/src/components/ElementView/ElementQueries.tsx

```diff
@@ -49,72 +72,110 @@ export const ElementQueries = () => {
+                actions.setSelected(null);
               }
-              actions.unBookmarkIntersection(bookmark.id, bookmark.label, bookmark.size);
+              actions.removeBookmark({ id: bookmark.id, label: bookmark.label, type: bookmark.type });
             }}
           />
         ))}
+        {/* Chip for the currently selected intersection */}
         {currentIntersection && !bookmarked.find((b) => b.id === currentIntersection.id) && (
-        <Chip
-          sx={(theme) => ({
-            margin: theme.spacing(0.5),
-            '.MuiChip-icon': {
-              color: nextColor,
-            },
-            backgroundColor: 'rgba(0,0,0,0.2)',
-          })}
-          icon={<SquareIcon fontSize={'1em' as any} />}
-          aria-label={`Selected intersection ${currentIntersectionDisplayName}, size ${currentIntersection.size}`}
-          onKeyDown={(e) => {
-            if (e.key === 'Enter') {
-              actions.bookmarkIntersection(
-                currentIntersection.id,
-                currentIntersectionDisplayName,
-                currentIntersection.size,
-              );
-            }
-          }}
-          label={`${currentIntersectionDisplayName} - ${currentIntersection.size}`}
-          onDelete={() => {
-            actions.bookmarkIntersection(
-              currentIntersection.id,
-              currentIntersectionDisplayName,
-              currentIntersection.size,
-            );
-          }}
-          deleteIcon={<StarBorderIcon />}
-        />
+          <Chip
+            sx={(theme) => ({
+              margin: theme.spacing(0.5),
+              '.MuiChip-icon': {
+                color: nextColor,
+              },
+              backgroundColor: 'rgba(0,0,0,0.2)',
+            })}
+            icon={<SquareIcon fontSize={'1em' as any} />}
+            aria-label={`Selected intersection ${currentIntersectionDisplayName}, size ${currentIntersection.size}`}
+            onKeyDown={(e) => {
+              if (e.key === 'Enter') {
+                actions.addBookmark<BookmarkedIntersection>({
+                  id: currentIntersection.id,
+                  label: currentIntersectionDisplayName,
+                  size: currentIntersection.size,
+                  type: 'intersection',
+                });
+              }
+            }}
+            label={`${currentIntersectionDisplayName} - ${currentIntersection.size}`}
+            onDelete={() => {
+              actions.addBookmark<BookmarkedIntersection>({
+                id: currentIntersection.id,
+                label: currentIntersectionDisplayName,
+                size: currentIntersection.size,
+                type: 'intersection',
+              });
+            }}
+            deleteIcon={<StarBorderIcon />}
+          />
+        )}
+        {/* Chip for the current element selection */}
+        {currentSelection?.selection && !bookmarked.find((b) => b.id === currentSelection.id) && (
```

Comment:
Do you have a helper you can use here in place of `currentSelection?.selection`

Reply:
No, should I add a selector for it?